SwiftReflector/MarshalEngineSwiftToCSharp.cs

```diff
@@ -32,8 +32,7 @@ public IEnumerable<ICodeElement> MarshalFunctionCall (FunctionDeclaration func,
 			postMarshalCode.Clear ();
 			ICodeElement returnLine = null;
 			var instanceEntity = typeMapper.GetEntityForTypeSpec (func.ParameterLists [0] [0].TypeSpec);
-			bool instanceIsProtocol = instanceEntity.EntityType == EntityType.Protocol;
-			//			bool returnIsGeneric = func.ReturnTypeSpec != null && func.IsGenericType(func.ReturnTypeSpec);
+			bool instanceIsProtocol = instanceEntity != null && instanceEntity.EntityType == EntityType.Protocol;
```

Comment:
You do not need to check if is null, you can do something like:
```suggestion
			bool instanceIsProtocol = instanceEntity?.EntityType == EntityType.Protocol;
```

The above like, if instanceEntity is null, will return null for EntityType which is not == to Protocol.